nucypher/blockchain/eth/token.py

```diff
@@ -596,19 +617,43 @@ def handle_working_errors(self, *args, **kwargs) -> None:
             self.log.warn('Unhandled error during work tracking: {failure.getTraceback()!r}',
                           failure=failure)
 
+    @property
+    def attempts(self) -> Tuple[int, int]:
+        return self.__success, self.__fail
+
     def __check_work_requirement(self) -> bool:
         # TODO: Check for stake expiration and exit
         if self.__requirement is None:
             return True
         try:
             r = self.__requirement()
             if not isinstance(r, bool):
-                raise ValueError(f"'requirement' must return a boolean.")
-        except TypeError:
-            raise ValueError(f"'requirement' must be a callable.")
+                raise WorkTracker.ProgrammingError(f"'requirement' function must return a boolean.")
+        except TypeError as e:
+            if not callable(self.__requirement):
+                raise WorkTracker.ProgrammingError(f"'requirement' must be a function or callable.") from e
+            raise
         return r
 
-    def _do_work(self) -> None:
+    def __make_commitment(self) -> bool:
+        transacting_power = self.worker.transacting_power
+        attempt = 0
+        while attempt < self.retry_attempts:
```

Comment:
@vepkenez @jMyles - You two have most recently engaged his area of the codebase.  Do you think the approach demonstrated here is now invalid (thinking of closing this PR for staleness)?